### DISCUSSION TITLE
Run script as user on the server

### DISCUSSION THREAD

**Discussion by bwicksall** [ltsp/ltsp#621] [Q&A] [answered]
On LTSP 5 I have multiple user templates and based on a users group I copy those templates over their profile at every login using:

`/usr/share/ldm/rc.d/S02-reset-user-profile`

Which calls:

`ssh -S ${LDM_SOCKET} ${LDM_SERVER} "/opt/ltsp-session.sh"`

Is there a supported way to do this in modern LTSP?  I'm using the LTSP 21.01 packages included in Debian Bullseye.

**Comment by alkisg** [comment, marked as answer]
The correct way would be from the server's /etc/pam.d/sshd. For example:

1. In `/etc/pam.d/sshd`, replace `@include common-session` with `@include common-session-sshd`.
2. Clone that file: `sudo cp /etc/pam.d/common-session /etc/pam.d/common-session-sshd`
3. Customize `/etc/pam.d/common-session-sshd` by adding this at its end:
   `session optional                        pam_exec.so     /usr/local/bin/local-common-session`
4. Create a `/usr/local/bin/local-common-session` shell script that does whatever you like, like the copy command you said. Start with `env > /tmp/pam-$$.env` to see what environment variables are available, like $PAM_USER.

That's all, now whenever a user runs `ssh server` or `sshfs server`, they'll get their customized home directory directly from the server, without having to customize their clients (or LTSP).
